### DISCUSSION TITLE
clicking location to copy met_location

### DISCUSSION THREAD

**Discussion by Ninjistix** [kwsch/PKHeX#3542] [Ideas]
in light of #3490, it takes too many steps to filter Pokémon in the encounter database, swapping back and forth from the encounter db and the batch editor.

Current Steps:
view Pokémon
ctrl m - batch editor
typing in met location to find value of =Location=
then opening the encounter database to type out =Location=XX with info retrieved from the batch editor

can we have an option of clicking met location to copy the =Location=XX value to the clipboard? like if you were to right or left even shift click on the met location, it would give you the data required for the encounter database, (the same could be done for other fields)

also it's impossible to use the batch editor and the encounter database at the same time, the batch editor window locks the pkhex window to that window, so you need to close the batch editor to go back into the encounter table.

**Comment by kwsch** [comment]
Clicking the met location is already used to suggest met location/level changes.

It might be better to add new syntax for the batch lines, to "use value from reference entity" instead.

However, this is problematic as the property names do not align between the two object types. Met_Location/Egg_Location are not Location/EggLocation etc.

Maybe it's best to extract the combo box+label displayed in the batch editor to a separate control, so you can quick lookup values when added to different forms.

**Reply by Ninjistix** [reply]
is it possible to make the batch editor window not lock up the UI? it would be helpful to search references from batch editor without having to close it and reopen it every time, or does it only apply to that specific pokemon in view?

**Reply by kwsch** [reply]
Batch Editor apples to party and boxes, but there's no real need to have it stay open for opening others.

I can just revise the Advanced tabs to show the batch editor controls there too.

**Reply by Ninjistix** [reply]
I feel like that would work, just having access to it from the editor might make things go faster, as far as finding specific Pokémon by location

**Comment by kwsch** [comment]
Added functionality in latest commit: https://github.com/kwsch/PKHeX/commit/70cab2aea77fbc3d7f6e1dac03249adac6cbe213

**Reply by Ninjistix** [reply]
just tested it, works fine, you just need to repopulate the filed for it to update the data

lets say I check a Met_Location for 1 Pokémon, it displays the results, but if I change Pokémon it doesn't update the results until I retype "Met_Location" again

**Reply by kwsch** [reply]
Yeah, there's no synchronization that lets the popup window know that the PKM Editor had its value changed.
